code/modules/mob/living/simple_animal/bot/firebot.dm

```diff
@@ -250,19 +249,16 @@
 
 //Look for burning people or turfs around the bot
 /mob/living/simple_animal/bot/firebot/process_scan(atom/scan_target)
-	var/result
-
 	if(scan_target == src)
-		return result
-
-	if(is_burning(scan_target))
-		if((detected_cooldown + DETECTED_VOICE_INTERVAL) < world.time)
-			speak("Fire detected!")
-			playsound(src, 'sound/voice/firebot/detected.ogg', 50, FALSE)
-			detected_cooldown = world.time
-		result = scan_target
-
-	return result
+		return src
+	if(!is_burning(scan_target))
+		return null
+
+	if((detected_cooldown + DETECTED_VOICE_INTERVAL) < world.time)
```

Comment:
Okay so this should use the cooldown macros

Reply:
This isn't firebot code improvement PR it'll happen when i get to them

Reply:
ok